packages/devextreme/js/ui/chat.d.ts

```diff
@@ -0,0 +1,182 @@
+import Widget, { WidgetOptions } from './widget/ui.widget';
+import {
+    Cancelable,
+    EventInfo,
+    NativeEventInfo,
+    InitializedEventInfo,
+    ChangedOptionInfo,
+} from '../events/index';
+
+/**
+ * @docid _ui_chat_ContentReadyEvent
+ * @public
+ * @type object
+ * @inherits EventInfo
+ */
+export type ContentReadyEvent = EventInfo<dxChat>;
+
+/**
+ * @docid _ui_chat_DisposingEvent
+ * @public
+ * @type object
+ * @inherits EventInfo
+ */
+export type DisposingEvent = EventInfo<dxChat>;
+
+/**
+ * @docid _ui_chat_InitializedEvent
+ * @public
+ * @type object
+ * @inherits InitializedEventInfo
+ */
+export type InitializedEvent = InitializedEventInfo<dxChat>;
+
+/**
+ * @docid _ui_chat_OptionChangedEvent
+ * @public
+ * @type object
+ * @inherits EventInfo,ChangedOptionInfo
+ */
+export type OptionChangedEvent = EventInfo<dxChat> & ChangedOptionInfo;
+
+/**
+ * @docid _ui_chat_MessageSendEvent
+ * @public
+ * @type object
+ * @inherits Cancelable,NativeEventInfo,Message
+ */
+export type MessageSendEvent = Cancelable & NativeEventInfo<dxChat, KeyboardEvent | PointerEvent | MouseEvent | TouchEvent> & Message;
+
+/**
+ * @docid
+ * @namespace DevExpress.ui.dxChat
+ * @public
+ */
+export type User = {
+    /**
+     * @docid
+     * @public
+     */
+    id: number;
+    /**
+     * @docid
+     * @public
+     */
+    firstName: string;
+    /**
+     * @docid
+     * @public
+     */
+    lastName: string;
+    /**
+     * @docid
+     * @public
+     */
+    avatarUrl?: string;
+};
+
+/**
+ * @docid
+ * @namespace DevExpress.ui.dxChat
+ * @public
+ */
+export type Message = {
+    /**
+     * @docid
+     * @public
+     */
+    timestamp: string;
+    /**
+     * @docid
+     * @public
+     */
+    author: User;
+    /**
+     * @docid
+     * @public
+     */
+    text: string;
+    /**
+     * @docid
+     * @public
+     */
+    typing?: boolean;
```

Comment:
Maybe "isTyping" would be a better name? I just didn't get what "typing" means from the first glance

Reply:
"typing" is a status that tells you whether the user is currently typing or not